src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -321,26 +303,77 @@ class TezosPlatformDiscoveryOperations(
     * @return list of attributes as a Future
     */
   def getTableAttributes(tableName: String): Future[Option[List[Attribute]]] = {
-    attributesCache.read.flatMap { entitiesMap =>
-      entitiesMap.get(tableName).map { case (last, attributes) =>
-        if (last + cacheTTL.toMillis > now) {
-          IO.pure(attributes)
-        } else {
-          for {
-            updatedAttributes <- IO.fromFuture(IO(getUpdatedAttributes(tableName, attributes)))
-            _ <- attributesCache.take
-            _ <- attributesCache.put(entitiesMap.updated(tableName, now -> updatedAttributes))
-          } yield updatedAttributes
-        }
-      }.sequence
-    }.unsafeToFuture()
+    caching.getCachingStatus.map { status =>
+      if (status == Finished) {
+        caching.getAttributes(tableName).flatMap { attributesOpt =>
+          attributesOpt.map { case (last, attributes) =>
+            if (last + cacheTTL.toMillis > now) {
+              IO.pure(attributes)
+            } else {
+              for {
+                updatedAttributes <- IO.fromFuture(IO(getUpdatedAttributes(tableName, attributes)))
+                _ <- caching.putAttributes(tableName, updatedAttributes)
+              } yield updatedAttributes
+            }
+          }.sequence
+        }.unsafeToFuture()
+      } else {
+        // if caching is not finished cardinality should be set to None
+        getTableAttributesWithoutUpdatingCache(tableName).map(_.map(_.map(_.copy(cardinality = None))))
+      }
+    }.unsafeToFuture().flatten
   }
 
   /** Runs query and attributes with updated counts */
   private def getUpdatedAttributes(tableName: String, columns: List[Attribute]): Future[List[Attribute]] = {
     metadataOperations.runQuery(getUpdatedAttributesQuery(tableName, columns))
   }
 
+  /** Returns current cache initialization status */
+  def getCacheStatus: Future[CachingStatus] =
+    caching.getCachingStatus.unsafeToFuture()
+
+  /** Inits attribute counts */
+  def initAttributesCount(): Unit = {
```

Comment:
Should we return a value so that it can be returned to the client to distinguish if the initialization was actually done or if the cache was already initialized?

Reply:
that's what I assumed `status` endpoint is for

Reply:
It could turn out useful anyway to have it here directly?
Or not exactly the same, it could be a (boolean?) return type that actually tells you if there was a need to initialize or not, so the caller could decide based on that, without the need to make a double check before and after the init call

Reply:
ok, now it returns current status

Reply:
I actually meant not to return the "current status", but if the "status changed", that is
- true if the values weren't there before and now they are
- false if nothing changed...

or even a pair of (oldStatus, newStatus), later wrapped as some json value to be inspected